modules/objdetect/src/qrcode.cpp

```diff
@@ -3709,7 +3708,15 @@ bool QRDetectMulti::checkSets(vector<vector<Point2f> >& true_points_group, vecto
     vector<int> set_size(true_points_group.size());
     for (size_t i = 0; i < true_points_group.size(); i++)
     {
-        set_size[i] = int( (true_points_group[i].size() - 2 ) * (true_points_group[i].size() - 1) * true_points_group[i].size()) / 6;
+        const std::uint64_t true_points_group_size = true_points_group[i].size();
+        CV_Assert(true_points_group_size - 2 <=
+                  std::numeric_limits<std::uint64_t>::max() / (true_points_group_size - 1));
+        CV_Assert((true_points_group_size - 2) * (true_points_group_size - 1) <=
+                  std::numeric_limits<std::uint64_t>::max() / true_points_group_size);
+        const std::uint64_t combination = (true_points_group_size - 2) *
+                                          (true_points_group_size - 1) * true_points_group_size / 6;
+        CV_Assert(combination <= std::numeric_limits<int>::max());
```

Comment:
3 asserts are overkill here. We have expression like (x-2)*(x-1)*x / 6. It means that single check `CV_Assert(x < limit)` or `CV_Assert(expr < std::numeric_limits<int>::max())` is enough.